In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
from bolides import BolideDataFrame;

C:\Users\under\anaconda3\Lib\site-packages\lightkurve\prf\__init__.py:7: UserWarning: Warning: the tpfmodel submodule is not available without oktopus installed, which requires a current version of autograd. See #1452 for details.
  warnings.warn(


In [3]:
bdf = BolideDataFrame(source='glm')
bdf

downloading: 22.8MiB [00:00, 33.5MiB/s]


,datetime,longitude,latitude,source,detectedBy,confidenceRating,lightcurveStructure,peak_energy_g17,peak_energy_g18,peak_energy_g19,peak_energy_cat_g16,peak_energy_cat_g17,peak_energy_cat_g18,peak_energy_cat_g19,_id,status,name,description,latitudeDelta,longitudeDelta,lastModifiedBy,lastModifiedDate,createdBy,createdDate,submittedBy,submittedDate,publishedBy,publishedDate,howFound,groundTrack,nearbyLightningActivity,otherDetectingSources,otherInformation,humanClassification,reason,rejectedBy,rejectedDate,duration,beginningAltitude,attachments,isManuallyGenerated,isPipelineIssue,files,location,checkout_time,assessmentScore,images,csv,platform,pdfFilenames,peak_energy_g16,geometry,phase,moon_fullness,solarhour,sun_alt_obs,sun_alt_app,date_retrieved
0,2025-07-13 17:48:37.901000+00:00,-130.898445,-25.096497,glm,"GLM-18,GLM-19",medium,good,NaN,0.0,0.0,,,Faint,Fairly Bright,68753096bd7bb0a35ac4be33,PUBLISHED,,,0.025391,0.856560,jdotson,2025-07-16T22:01:39.769Z,auto,2025-07-14T09:30:14.369Z,rlongenb,2025-07-14T21:19:52.763Z,jdotson,2025-07-16T22:01:39.769Z,"{'GLM-19': ['forest'], 'GLM-18': ['forest']}","{'GLM-18': {'category': 'Multi-Pixel', 'value'...",isolated,,,Bolide,,,None,0.466926,84.781978,[{'location': {'coordinates': [-131.7550048828...,False,False,[G19_2025-07-13_17h48m37.947_-25.111_-131.585_...,"{'coordinates': [-130.89844512939453, -25.0964...",10,"{'score': 0.9870747327804565, 'assessment': 'a...","[{'name': 'Map', 'url': '/img/OR_GLM-L2-LCFA_G...",[/csv/OR_GLM-L2-LCFA_G19_s20251941748200_e2025...,"GLM-18,GLM-19",[G19_2025-07-13_17h48m37.947_-25.111_-131.585_...,NaN,POINT (-130.89845 -25.0965),0.623400,0.753200,8.986386,25.876689,25.844012,2025-07-18 12:37:29.234328
1,2025-07-05 18:14:49.855000+00:00,-132.722153,1.872810,glm,GLM-18,low,good,NaN,0.0,NaN,,,Faint,,68702785676cc8e13b2449a0,PUBLISHED,,,0.035545,0.001022,jdotson,2025-07-16T22:01:54.573Z,auto,2025-07-10T13:50:13.042Z,rlongenb,2025-07-14T23:46:32.759Z,jdotson,2025-07-16T22:01:54.573Z,{'GLM-18': ['forest']},"{'GLM-18': {'category': 'Multi-Pixel', 'value'...",isolated,,,Bolide,,,None,0.216678,,[{'location': {'coordinates': [-132.7231750488...,False,False,[G18_2025-07-05_18h14m49.838_1.909_-132.722_18...,"{'coordinates': [-132.72215270996094, 1.872810...",6,"{'score': 0.1728009730577469, 'assessment': 'c...","[{'name': 'Map', 'url': '/img/OR_GLM-L2-LCFA_G...",[/csv/OR_GLM-L2-LCFA_G18_s20251861814400_e2025...,GLM-18,[G18_2025-07-05_18h14m49.838_1.909_-132.722_18...,NaN,POINT (-132.72215 1.87281),0.351548,0.703096,9.320244,45.812743,45.797334,2025-07-18 12:37:29.234328
2,2025-07-05 13:08:34.016000+00:00,-53.601509,3.543533,glm,GLM-19,low,good,NaN,NaN,0.0,,,,Faint,68702784676cc8e13b244940,PUBLISHED,,,0.000000,0.000000,jdotson,2025-07-16T22:02:12.020Z,auto,2025-07-10T13:50:12.795Z,rlongenb,2025-07-14T23:18:50.407Z,jdotson,2025-07-16T22:02:12.020Z,{'GLM-19': ['forest']},"{'GLM-19': {'category': 'Single-Pixel', 'value...",isolated,,,Bolide,,,None,0.055696,,[{'location': {'coordinates': [-53.60150909423...,False,False,[G19_2025-07-05_13h08m34.016_3.543_-53.601_192...,"{'coordinates': [-53.60150909423828, 3.5435333...",6,"{'score': 0.8912568092346191, 'assessment': 'c...","[{'name': 'Map', 'url': '/img/OR_GLM-L2-LCFA_G...",[/csv/OR_GLM-L2-LCFA_G19_s20251861308200_e2025...,GLM-19,[G19_2025-07-05_13h08m34.016_3.543_-53.601_192...,NaN,POINT (-53.60151 3.54353),0.344304,0.688608,9.491160,48.859246,48.845398,2025-07-18 12:37:29.234328
3,2025-07-05 11:54:41.174000+00:00,-37.083141,-46.501108,glm,GLM-19,low,good,NaN,NaN,0.0,,,,Fairly Bright,68702784676cc8e13b244941,PUBLISHED,,,0.046721,0.046719,jdotson,2025-07-16T22:02:45.854Z,auto,2025-07-10T13:50:12.798Z,rlongenb,2025-07-14T23:18:26.793Z,jdotson,2025-07-16T22:02:45.854Z,{'GLM-19': ['forest']},"{'GLM-19': {'category': 'Multi-Pixel', 'value'...",isolated,,,Bolide,,,None,0.193027,,[{'location': {'coordinates': [-37.12985992431...,False,False,[G19_2025-07-05_11h54m41.175_-46.548_-37.039_1...,"{'coordinates': [-37.0831413269043, -4

In [4]:
highbdf= bdf[bdf.confidenceRating == "high"]

In [5]:
latdelta=highbdf.latitudeDelta #isolate latitude and longitude change 
longdelta=highbdf.longitudeDelta

In [6]:
print(np.max(latdelta), np.max(longdelta))

15.681482315063477 18.090576171875


In [7]:
delta= np.sqrt(latdelta**2+longdelta**2) #use the pythagorean theorem to compute the actual track length of the bolides

In [8]:
np.max(delta)

23.941132676380626

In [9]:
bolidelist=[] #list of the longest tracked bolides 
latdeltalist=[] #the corresponding change in latitude for the longest tracked bolides
longdeltalist=[] #corresponding change in longitude, this will be used later to find the bolide locations in highbdf
indexlist=[] #tracks the placement of the bolides selected in delta

for i in range(len(delta)):   
    if delta.iloc[i]>3: #arbitrary number that shows the bolides with the greatest delta
        bolidelist.append(delta.iloc[i]) #append all values to their corresponding lists 
        latdeltalist.append(latdelta.iloc[i])
        longdeltalist.append(longdelta.iloc[i])
        indexlist.append(i)

bolidelist

[23.941132676380626,
 3.1622776601683795,
 3.1622776601683795,
 3.605551275463989,
 3.605551275463989,
 3.605551275463989,
 3.1622776601683795,
 3.605551275463989,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.605551275463989,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.605551275463989,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 4.47213595499958,
 3.1622776601683795]

In [10]:
len(latdeltalist)

24

In [11]:
bolidenumbers=[] #will hold the locations in highbdf of the bolides we want
w=0 
for i in range(len(highbdf)):
    for e in range (len(latdeltalist)): #search latdeltalist and longdeltalist for the corresponding values...
        if latdelta.iloc[i] == latdeltalist[e] and longdelta.iloc[i] == longdeltalist[e] and i not in bolidenumbers: #... and compare those values to make sure that specific bolide is the one we want
             bolidenumbers.append(i) #save the index 

We need to remove duplicates so we have 15 bolides. I'm going to take out 9 who have the value 3.16..., the indexes of which are 1, 2, 6, 8, 9, 10, 11, 12, 14

In [12]:
bolidenumbers #pre removal- len=24

[79,
 220,
 404,
 433,
 442,
 471,
 481,
 482,
 486,
 490,
 509,
 511,
 512,
 531,
 570,
 583,
 595,
 598,
 626,
 631,
 633,
 641,
 661,
 670]

In [13]:
bolidenumbers.remove(bolidenumbers[1]) #remove some shorter tracked bolides so our sample size is 15
bolidenumbers.remove(bolidenumbers[2]) #this removal is ONLY for the sample size, it's possible the 3D trajectories for these are quite interesting
bolidenumbers.remove(bolidenumbers[6]) #but for now they didn't make the cut
bolidenumbers.remove(bolidenumbers[8]) 
bolidenumbers.remove(bolidenumbers[9])
bolidenumbers.remove(bolidenumbers[10])
bolidenumbers.remove(bolidenumbers[11])
bolidenumbers.remove(bolidenumbers[12])
bolidenumbers.remove(bolidenumbers[14])

In [14]:
totalbolidelist=[]

for i in range (len(bolidenumbers)):
    totalbolidelist.append(highbdf.iloc[bolidenumbers[i]]) #make a list of the information regarding the bolides we want

In [15]:
bolidenumbers #final location of the 15 selected bolides within highbdf

[79, 404, 442, 471, 481, 482, 490, 509, 512, 570, 595, 626, 633, 641, 670]

In [16]:
totalbolidelist #all the information regarding every bolide selected

[datetime                                    2024-09-19 09:52:50.049000+00:00
 longitude                                                         -95.642914
 latitude                                                          -11.445276
 source                                                                   glm
 detectedBy                                                            GLM-16
 confidenceRating                                                        high
 lightcurveStructure                                                very good
 peak_energy_g17                                                          NaN
 peak_energy_g18                                                          NaN
 peak_energy_g19                                                          NaN
 peak_energy_cat_g16                                                   Bright
 peak_energy_cat_g17                                                         
 peak_energy_cat_g18                                            